subsys/fb/cfb.c

```diff
@@ -105,24 +105,39 @@ static uint8_t draw_char_vtmono(const struct char_framebuffer *fb,
 	}
 
 	for (size_t g_x = 0; g_x < fptr->width; g_x++) {
-		uint32_t y_segment = y / 8U;
-
-		for (size_t g_y = 0; g_y < fptr->height / 8U; g_y++) {
-			uint32_t fb_y = (y_segment + g_y) * fb->x_res;
-			uint8_t byte;
-
-			if ((fb_y + x + g_x) >= fb->size) {
-				return 0;
+		int16_t fb_x = x + g_x;
+		uint8_t byte = get_glyph_byte(glyph_ptr, fptr, g_x, 0);
+
+		for (size_t g_y = 0; g_y < fptr->height; g_y++) {
+			int16_t fb_y = y + g_y;
+			size_t fb_index = (fb_y / 8U) * fb->x_res + fb_x;
```

Comment:
Would be nice to have short and concise comments on few points, otherwise no one can understand what happens at first glance.

Reply:
Add explaining the purpose of each in the process branching.